### PR TITLE
bump minimum Armadillo version

### PR DESCRIPTION
**Pull request by conradsnicta**
Update ensmallen webpage to state we now need Armadillo 10.8 and C++14.

Companion to https://github.com/mlpack/ensmallen/pull/400 and https://github.com/mlpack/ensmallen/pull/404

### CODE REVIEW

**Review by eddelbuettel — APPROVED** [review]
:ship:

**Review by shrit — APPROVED** [review]

**Review by rcurtin — APPROVED** [review]
Nice, looks great.  I thought about waiting until the next ensmallen release to merge this, but I don't think it matters enough, better to just get it done with now.

**Comment by eddelbuettel** [comment]
> better to just get it done with now.

I found over the years that it works well to treat the git repo as a rolling rc.

**Pull request merged**
